commands/download.js

```diff
@@ -0,0 +1,76 @@
+var chalk = require('chalk')
+var prettyBytes = require('pretty-bytes')
+var Dat = require('../lib/dat')
+var logger = require('../lib/logger')
+var ui = require('../lib/ui')
+
+module.exports = function (args) {
+  var dat = Dat(args)
+  var log = logger(args)
+
+  var downloadTxt = 'Downloading '
+  var swarmTimeout = null
+
+  log.status('Starting Dat...\n', 0)
+  log.status('Connecting...', 1)
+
+  dat.on('error', onerror)
+
+  dat.on('ready', function () {
+    log.message('Downloading in ' + dat.dir + '\n')
+    dat.download(function (err) {
+      if (err) onerror(err)
```

Comment:
does the callback in  `dat.download(function (err) { })` fire when the download is finished or only when there is an error?